src/modules/metrics/actions.js

```diff
@@ -22,11 +21,23 @@ export function addMetric(item) {
   }
 }
 
-//spaceId must be done before the metric is removed here.
 export function removeMetrics(ids) {
   return (dispatch, getState) => {
     if (ids.length === 0) { return }
+
     const spaceId = findSpaceId(getState, ids[0])
+
+    const {organizations, spaces, facts: {organizationFacts}} = getState()
+    const organizationId = e.collections.gget(spaces, spaceId, 'id', 'organization_id')
+    if (!!organizationId) {
+      const organization = e.collections.get(organizations, organizationId)
+      const facts = e.organization.findFacts(organizationId, organizationFacts)
+      const factsToDelete = _.filter(facts, f => f.exported_from_id === spaceId && ids.includes(f.metric_id))
+      factsToDelete.forEach(fact => {
```

Comment:
I agree, but soon I'd like to refactor the facts client side storage (get rid of the organizationFacts middle layer and just store facts by ID like everything else) and this section will undergo heavy changes then that will make it more amenable to such a change, so I'd prefer to make that change then.